app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,78 @@
+class RentalsController < ApplicationController
+
+  def checkout
+    existing_customer = Customer.find_by(id: params[:customer_id])
+    existing_video = Video.find_by(id: params[:video_id])
+
+    if existing_customer.nil? || existing_video.nil?
+      render json: {
+          errors: ['Not Found']
+      }, status: :not_found
+      return
+    elsif existing_video.available_inventory <= 0
+      render json: {
+          errors: ['Bad Request']
+      }, status: :bad_request
+      return
+    end
+
+    rental = Rental.new(rental_params)
+
+    # create a new rental
+    if rental.save
+
+      rental.increase_customer_video_count
+      rental.decrease_available_video_inventory
+      rental.return_date
+
+      customer = Customer.find_by(id: existing_customer.id)
+      video = Video.find_by(id: existing_video.id)
+
+      render json: {
+          customer_id: rental.customer_id,
+          video_id: rental.video_id,
+          due_date: rental.due_date,
```

Comment:
This is the business logic! Instead of breaking out small chunks to the model, put the whole of this action into the model!